packages/accounts-base/accounts_client.js

```diff
@@ -63,6 +66,31 @@ export class AccountsClient extends AccountsCommon {
   }
 
   /**
+   * @summary Register a new login function on client. Intended for OAuth package authors. You can call the login function by using `Accounts.callLoginFunction`.
+   * @locus Client
+   * @param {String} funcName The name of your login function. Used by `Accounts.callLoginFunction`. Should be the OAuth provider name accordingly.
+   * @param {Function} func The actual function you want to call. Just write it in the manner of `loginWithFoo`.
+   */
+  registerClientLoginFunction(funcName, func) {
+    if (this._loginFuncs[funcName]) {
+      throw new Error(`${funcName} has been defined already`);
+    }
+    this._loginFuncs[funcName] = func;
+  }
+  /**
+   * @summary Call a login function defined using `Accounts.registerClientLoginFunction`.
+   * @locus Client
+   * @param {String} funcName The name of the login function you wanted to call.
+   * @param {Array} funcArgs The `arguments` for the login function.
+   */
+  callLoginFunction(funcName, funcArgs) {
```

Comment:
Could we have both `callLoginFunction` (which takes multiple arguments) and `applyLoginFunction` (which takes an array of arguments), similar to `Function.prototype.{call,apply}`?